quelpa.el

```diff
@@ -1606,13 +1624,27 @@ Return t in each case."
 
 (defun quelpa-update-cache (cache-item)
   ;; try removing existing recipes by name
-  (setq quelpa-cache (cl-remove (car cache-item)
-                                quelpa-cache :key #'car))
+  (setq quelpa-cache (cl-remove-if
+                      (lambda (item) (eq (car cache-item) (car item)))
+                      quelpa-cache))
   (push cache-item quelpa-cache)
   (setq quelpa-cache
         (cl-sort quelpa-cache #'string<
                  :key (lambda (item) (symbol-name (car item))))))
 
+(defun quelpa-remove-cache (cache-item)
+  "Remove cache-item from `quelpa-cache'."
```

Comment:
The name of this function suggests that it removes the cache itself rather than removing an item from the cache.  How about something like `quelpa-cache-remove-item`?